index.js

```diff
@@ -1,55 +1,314 @@
+function createStore(item) {
+  const ul = document.querySelector(".store--item-list")
+  const li = document.createElement("li")
+
+  const div = document.createElement("div")
+  div.setAttribute("class", "store--item-icon")
+
+  const img = document.createElement("img")
+  img.setAttribute("src", `assets/icons/${item.id}.svg`)
+
+  const name = document.createElement("span")
+  name.setAttribute("class", "store--item-name")
+  name.innerText = item.name
+
+  const price = document.createElement("span")
+  price.setAttribute("class", "store--item-price")
+  price.innerText = `£${item.price}`
+
+  const button = document.createElement("button")
+  button.innerText = "ADD TO CART"
+  button.addEventListener("click", () => {
+    addToCart(item)
+  })
+
+  ul.appendChild(li)
+  li.appendChild(div)
+  li.appendChild(name)
+  li.appendChild(price)
+  li.appendChild(button)
+  div.appendChild(img)
+}
+// render and append individual item to cart
+function createCart(item) {
+  const ul = document.querySelector(".cart--item-list")
+  const li = document.createElement("li")
+
+  const img = document.createElement("img")
+  img.setAttribute("class", "cart--item-icon")
+  img.setAttribute("src", `assets/icons/${item.id}.svg`)
+  img.setAttribute("alt", item.name)
+
+  const p = document.createElement("p")
+  p.innerText = item.name
+
+  const span = document.createElement("span")
+  span.setAttribute("class", "quantity-text center")
+  span.innerText = item.quantity
+  
+  const buttonRemove = document.createElement("button")
+  buttonRemove.setAttribute("class", "quantity-btn remove-btn center")
+  buttonRemove.innerText = "-"
+  buttonRemove.addEventListener("click", () => {removeFromCart(item)})
+  
+  const buttonAdd = document.createElement("button")
+  buttonAdd.setAttribute("class", "quantity-btn add-btn center")
+  buttonAdd.innerText = "+"
+  buttonAdd.addEventListener("click", () => {addToCart(item)})
+
+  ul.appendChild(li)
+  li.appendChild(img)
+  li.appendChild(p)
+  li.appendChild(buttonRemove)
+  li.appendChild(span)
+  li.appendChild(buttonAdd)
+}
+
+// Called when Shop or Cart add button pressed
+function addToCart(item) {
+  const itemIndex = data.cart.indexOf(item)
+  // if item exists, update
+  if (data.cart.includes(item)) {
+    data.cart[itemIndex].quantity += 1
+  }
+  // if item dont exists, add quantity and push
+  else {
+    item.quantity = 1
+    data.cart.push(item)
+  }
+
+  renderCart()
+}
+function removeFromCart(item) {
+  const itemIndex = data.cart.indexOf(item)
+  // if quantity is more than 1, subtract, if 1 then remove completely
+  if (item.quantity > 1) {
+    data.cart[itemIndex].quantity -= 1
+  }
+  else {
+    data.cart.splice(itemIndex, 1)
+  }
+
+  renderCart()
+}
+
+// Iterates though the Cart items checking for the item quantity.
+function updateTotal() {
+  const totalSpan = document.querySelector(".total-number")
+  let total = 0
+
+  data.cart.forEach((item) => {
```

Comment:
You could try using the awesome and confusing reduce function to do the same thing :)